### PR TITLE
refactor(switch): signal migration

### LINKED ISSUE #438: refactor(switch): migration to signals
### Which scope/s are relevant/related to the feature request?

switch

### Information

Migrate away from all Input/Output and View Decorators

### Describe any alternatives/workarounds you're currently using

_No response_

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by ashley-hunter**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows our
      guidelines: https://github.com/goetzrobin/spartan/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] accordion
- [ ] alert
- [ ] alert-dialog
- [ ] aspect-ratio
- [ ] avatar
- [ ] badge
- [ ] button
- [ ] calendar
- [ ] card
- [ ] checkbox
- [ ] collapsible
- [ ] combobox
- [ ] command
- [ ] context-menu
- [ ] data-table
- [ ] date-picker
- [ ] dialog
- [ ] dropdown-menu
- [ ] hover-card
- [ ] icon
- [ ] input
- [ ] label
- [ ] menubar
- [ ] navigation-menu
- [ ] pagination
- [ ] popover
- [ ] progress
- [ ] radio-group
- [ ] scroll-area
- [ ] select
- [ ] separator
- [ ] sheet
- [ ] skeleton
- [ ] slider
- [ ] sonner
- [ ] spinner
- [X] switch
- [ ] table
- [ ] tabs
- [ ] textarea
- [ ] toast
- [ ] toggle
- [ ] tooltip
- [ ] typography

## What is the current behavior?

Closes #438 

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: libs/ui/core/src/lib/brain/rx-host-integration.ts [hunk 1]
```diff
@@ -58,5 +58,5 @@ export function rxHostPressedListener() {
 			}),
 			filter(Boolean),
 		),
-	).pipe(debounceTime(0));
```
[the remark this answers is not in the document]
Comment (goetzrobin) [reply libs/ui/core/src/lib/brain/rx-host-integration.ts:61]: How are you feeling about rxHostPressedListener? That was a super early implementation and not sure if it's worth the upkeep!

[the remark this answers is not in the document]
Comment (ashley-hunter) [reply libs/ui/core/src/lib/brain/rx-host-integration.ts:61]: Yeah, I agree, personally I think it's easier and more obvious just to add a two hostlisteners. Saves jumping between files and it's not immediately obvious what this functions purpose is. I don't really see the benefit in it myself.

[the remark this answers is not in the document]
Comment (goetzrobin) [reply libs/ui/core/src/lib/brain/rx-host-integration.ts:61]: It was supposed to be advanced into something like usePress from React Aria, which is great for mobile apps. But I never got around to that. Happy to deprecate/remove before v1

[the remark this answers is not in the document]
Comment (ashley-hunter) [reply libs/ui/core/src/lib/brain/rx-host-integration.ts:61]: Makes sense. If that is ever something you want to bring in, I ported all those interactions from react aria to angular for angular primitives, so they could easily be carried over to Spartan.

[the remark this answers is not in the document]
Comment (ashley-hunter) [reply libs/ui/core/src/lib/brain/rx-host-integration.ts:61]: I have now removed the rxHostListeners from all components.

**Pull request merged**